### PR TITLE
add price increase during tokensale

### PR DESCRIPTION
**Pull request by CJentzsch**
extra ether are transferred to another account. They can be send to the DAO after they have spent at least this amount of ether.

### CODE REVIEW

File: TokenSale.sol [hunk 1]
```diff
@@ -85,12 +91,24 @@ contract TokenSale is TokenSaleInterface, Token {
...
+            if (msg.sender.call.value(weiGiven[msg.sender])()) {
+                Refund(msg.sender, weiGiven[msg.sender]);
+                totalSupply -= balances[msg.sender];
+                balances[msg.sender] = 0;
+            }
         }
     }
+
+    function divisor() returns (uint divisor){
+        if (closingTime - 2 weeks > now) return 20;
```
Comment (LefterisJP) [inline TokenSale.sol]: Definitely add comments here in all 3 legs of the if explaining the different periods and the pricing increase.

File: TokenSale.sol [hunk 2]
```diff
@@ -85,12 +91,24 @@ contract TokenSale is TokenSaleInterface, Token {
 
     function refund() noEther {
         if (now > closingTime
-            && !isFunded
-            && msg.sender.call.value(balances[msg.sender])()) // execute refund
+            && !isFunded)
```
Comment (LefterisJP) [inline TokenSale.sol]: you can now put the if back in one line :), no longer exceeds the 80 column limit

File: TokenSale.sol [hunk 3]
```diff
@@ -33,7 +33,11 @@ contract TokenSaleInterface {
     bool public isFunded;   
     // For DAO splits - if privateSale is 0, then it is a public sale, otherwise
     // only the address stored in privateSale is allowed to purchase tokens
-    address public privateSale;               
+    address public privateSale;
+    // used to store extra ether which has been paid after the price has increased
```
Comment (LefterisJP) [inline TokenSale.sol]: after the DAO *token* price has increased

File: DAO.sol [hunk 4]
```diff
@@ -657,6 +664,8 @@ contract DAO is DAOInterface, Token, TokenSale {
         if (_recipient == serviceProvider 
             || _recipient == address(rewardAccount) 
             || _recipient == address(this)
+            || (_recipient == address(extraBalance)
```
Comment (LefterisJP) [inline DAO.sol]: Please write a comment here (on top, not next to it ^_^ ) explaining that the DAO can only receive ether from the extraBalance account only once they have already spent as much as that account's balance is.

Comment (CJentzsch) [inline DAO.sol]: done

File: DAO.sol [hunk 5]
```diff
@@ -657,6 +664,10 @@ contract DAO is DAOInterface, Token, TokenSale {
...
+                // only allowed when at least the amount hold in the
```
Comment (LefterisJP) [inline DAO.sol]: hold -> held

File: TokenSale.sol [hunk 6]
```diff
@@ -84,13 +90,29 @@ contract TokenSale is TokenSaleInterface, Token {
     }
 
     function refund() noEther {
-        if (now > closingTime
-            && !isFunded
-            && msg.sender.call.value(balances[msg.sender])()) // execute refund
+        if (now > closingTime && !isFunded)
```
Comment (LefterisJP) [inline TokenSale.sol]: also move the `{` up at the right of the parentheses

**Pull request merged**